File: 005/in_triangle.rb
def S_triangle(x1, y1, x2, y2, x3, y3)
  0.5 * (x1*(y2 - y3) + x2*(y3 - y1) + x3*(y1 - y2)).abs
end

def check_in(s1, s2, s3, sf)
  st = s1 + s2 + s3
  if st == sf
    if s1 == 0 || s2 ==0 || s3 == 0

      puts "Diem nam tren tam giac"
    
    else
    
      puts "Diem nam trong tam giac"
    
    end
  else
    puts "Diem nam ngoai tam giac"
  end

end

File: 005/in_triangle.c
#include "in_triangle.h"

double S_triangle(double x1, double y1, double x2, double y2, double x3, double y3) {
	return 0.5 * fabs(x1*(y2 - y3) + x2*(y3 - y1) + x3*(y1 - y2));
}

void check_in(double S1, double S2, double S3, double Sf) {
	double St = S1 + S2 + S3;
	if (St == Sf) {
		if (S1 == 0 || S2 == 0 || S3 == 0) {
			printf("Diem nam tren canh tam giac\n");
		} else {
			printf("Diem nam trong tam giac\n");
		}
	} else {
		printf("Diem nam ngoai tam giac\n");
	}
}
